sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java

```diff
@@ -1280,6 +1288,111 @@ public void testGetInboundMessagesByBodyTypeAndSequenceId(@TempDir final File di
         }
     }
 
+    /**
+     * Tests whether getInboundMessageByBodyTypeAndSequenceId() orders the results by MdibVersion and Timestamp
+     *
+     * @param dir message storage directory
+     * @throws IOException on io exceptions
+     */
+    @Test
+    public void testGetInboundMessagesByBodyTypeAndSequenceIdGoodCheckOrderedByMdibVersionAndTimestamp(@TempDir final File dir)
```

Comment:
To increase readibility i would suggest removing all unnecessary messages in this unit test. Just add a few messages which are identical except for the timestamp to the storage and verify the first you encounter is the one with the lowest timestamp.

Reply:
ok

Reply:
Done.

Reply:
To make sure that sorting really takes place, I would suggest not storing the messages in the MessageStorage in the same order in which they are to be output.

Reply:
Done.